pyat/at/physics/dynamic_aperture.py

```diff
@@ -0,0 +1,997 @@
+                             'ct': 1
+                             }
+        elif mode == 'delta-x':
+            self.n_points = {'x': 13,
+                             'xp': 1,
+                             'y': 1,
+                             'yp': 1,
+                             'delta': 13,
+                             'ct': 1
+                             }
+        elif mode == 'ct-delta':
+            self.n_points = {'x': 1,
+                             'xp': 1,
+                             'y': 1,
+                             'yp': 1,
+                             'delta': 13,
+                             'ct': 13
+                             }
+        elif mode == 'x-xp':
+            self.n_points = {'x': 13,
+                             'xp': 13,
+                             'y': 1,
+                             'yp': 1,
+                             'delta': 1,
+                             'ct': 1
+                             }
+        elif mode == 'y-yp':
+            self.n_points = {'x': 1,
+                             'xp': 1,
+                             'y': 13,
+                             'yp': 13,
+                             'delta': 1,
+                             'ct': 1
+                             }
+        elif mode == 'xp-yp':
+            self.n_points = {'x': 1,
+                             'xp': 13,
+                             'y': 1,
+                             'yp': 13,
+                             'delta': 1,
+                             'ct': 1
+                             }
+        elif mode == '6D':
+            self.n_points = {'x': 5,
+                             'xp': 5,
+                             'y': 5,
+                             'yp': 5,
+                             'delta': 5,
+                             'ct': 5
+                             }
+        else:  # default for no mode = x-y
+            self.n_points = {'x': 13,
+                             'xp': 1,
+                             'y': 13,
+                             'yp': 1,
+                             'delta': 1,
+                             'ct': 1
+                             }
+
+        self.test_points_grid(compute_limits=compute_range)
+
+        pass
+
+    def compute_orbit(self):
+        """
+        computes orbit for given dp
+        """
+
+        if self.ring.radiation:
+            # set RF frequency
+            drf = - self.rf_frequency * self.mcf * self.dp
+
+            new_rf_frequency = self.rf_frequency + drf
+            if self.verbose:
+                print('dp = {dp}, rf set to f_RF0 + {rf} Hz'.format(dp=self.dp, rf=drf))
+                print('new RF = {rf:3.6f} MHz'.format(rf=new_rf_frequency*1e-6))
+
+            # set RF frequency to all cavities
+            try:
+                self.ring = self.ring.set_rf_frequency(new_rf_frequency, copy=True)
+            except Exception:
+                print('RF not set. Probably the lattice has no RFCavity.')
```

Comment:
Do you really want to continue after this error ? With radiation but no cavity, there is no hope to get a closed orbit. Don't catch the exception!

Note: in such a case, you should catch the most accurate exception (`AtError` in that case), to avoid catching any other one (memory fault...) that you are unable to handle.

Reply:
like this ? 

```
            try :
                self.ring = self.ring.set_rf_frequency(new_rf_frequency, copy=True)
            except AtError as exc:
                print('RF not set. Probably the lattice has no RFCavity.')
                raise(exc)
```